app/ts/utils/ethereumUtils.ts

```diff
@@ -46,3 +48,45 @@ export function isSameAddress(address1: `0x${ string }` | undefined, address2: `
 	if (address1 === undefined || address2 === undefined) return false
 	return address1.toLowerCase() === address2.toLowerCase()
 }
+
+export const bigintToNumberFormatParts = (amount: bigint, decimals = 18n, maximumSignificantDigits = 4) => {
+	const floatValue = Number(formatUnits(amount, Number(decimals)))
+
+	let formatterOptions: Intl.NumberFormatOptions = { useGrouping: false, maximumFractionDigits: 3 }
+
+	// maintain accuracy if value is a fraction of 1 ex 0.00001
+	if (floatValue % 1 === floatValue) formatterOptions.maximumSignificantDigits = maximumSignificantDigits
+
+	// apply only compacting with prefixes for values >= 10k or values <= -10k
+	if (Math.abs(floatValue) >= 1e4) {
+		formatterOptions = { minimumFractionDigits: 0, notation: 'compact' }
+	}
+
+	const formatter = new Intl.NumberFormat('en-US', formatterOptions)
+	const parts = formatter.formatToParts(floatValue)
+	const partsMap = new Map<Intl.NumberFormatPartTypes, string>()
+
+	for (const part of parts) {
+		if (part.type === 'compact') {
+			// replace American format with Metric prefixes https://www.ibiblio.org/units/prefixes.html
+			const prefix = part.value.replace('K', 'k').replace('B', 'G')
+			partsMap.set(part.type, prefix)
+			continue
+		}
+		partsMap.set(part.type, part.value)
+	}
+
+	return partsMap
+}
```

Comment:
I don't think we should *ever* use JavaScript `number` for anything other than small values for things like dates and such.  Anything that touches crypto should not be a `number` ever, either a string or a bigint.  Can you convince me that this is not a really bad idea to include?

We already have `bigintToDecimalString` and `stringToDecimalBigint`, why do we need this function?

Reply:
`bigintToDecimalString` produces insane big numbers if there's decimals. This function rounds them, which I feel is really handy for displaying values

Reply:
I'm fine with rounding, but we should not be using `number` for it.  Simple truncation works in many cases, and we can do the math to correctly round if we want to round at 5.

Reply:
yeah. This code is also in interceptor, I guess it could be fixed in both

Reply:
Oh, yes.  We shouldn't be using `number` in interceptor either.  Too easy to get it wrong and display incorrect value to the user.

Reply:
I removed this code for now as I couldn't get it to work as I wanted and its not so critical to have for now